docs/UserGuide.md

```diff
@@ -116,19 +116,107 @@ Examples:
 - `edit 1 p/999` will overwrite the 1st client's phone number to `999`.
 - `edit 2 n/Kylie  i/3333 f/5` will overwrite the 2nd client's name to `Kylie`, income to `3333` and family size to `5`.
 
-### Filtering clients : `filter`
-
-Filters the list of client with an input keyphrase.
-
-Format: `filter KEYPHRASE`
-
-- The search is case-insensitive. e.g `james` will match `James`
-- Partial words will still be matched e.g. `Udh` will match `Udhaya`
-- All persons' names containing the keyword will be returned e.g. `Al` will return `Alicia`, `Allysa`
-
-Example:
-
-- `filter Al` will list out persons whose name has `"Al"` inside, such as `"Alicia"`, `"Allysa"` and `"Jamal"`
+### Filtering clients: `filter`
+
+The filter command in Realodex allows you to narrow down your list of clients by specifying a keyphrase 
+related to one of the client's attributes: name, remark, tag, birthday month, or housing type.
+This feature is particularly useful when you need to focus on a
+subset of your client database that meets certain criteria.
+
+>
+> When performing multiple filter operations in sequence, 
+> each new filter is applied to the original, full list of clients, not the subset produced by the previous filter. 
+> This approach ensures clarity and consistency in search results
+
+<u>Filter by Name format:</u> `filter n/KEYPHRASE`
+
+- Returns the list of clients whose names contain the specified keyphrase.
+- Keyphrase input should be in valid format for names (alphanumeric) and non-empty.
+- The search is **case-insensitive**.
+  - `filter n/james` matches person with the name "James".
+- **Partial fragments** of names will still be matched.
+  - `filter n/Udh` matches a person with the name "Udhaya".
+- **Comprehensive searching**, returning all persons with names containing the keyphrase.
+  - `filter n/Al` returns persons named "Alicia", "Allysa", "Jamal".
+  
+![filterByNameScreenshot.png](images/filter/filterByNameScreenshot.png)
+<p align="center">
+  <em> <code>filter n/Li</code> returns persons with names like "Charlotte Oliveiro" and "David Li"</em>
+</p>
+
+<u>Filter by Remark format:</u> `filter r/KEYPHRASE`
+
+- Returns the list of clients whose remarks include the specified keyphrase.
+- Keyphrase input should be non-empty. 
+This is an intentional design choice to ensure that the command is used for targeted searches, preventing the potential misinterpretation of an empty keyphrase as a request to list all clients.
+- Important: The remarks for the `filter r/` command must not contain any other prefixes to prevent parsing errors. 
+> The command `filter r/ my tag is t/buyer` would cause an error because the system interprets `t/` as the start of a new prefix.
+> To avoid this, ensure that the remark does not contain any spaces followed by slashes that could be misconstrued as additional prefixes.
```

Comment:
Might need to add a link pointing to the to input constraints summary that @4llysa  will add, just KIV